io/io/src/TDirectoryFile.cxx

```diff
@@ -1186,17 +1186,21 @@ TFile *TDirectoryFile::OpenFile(const char *name, Option_t *option,const char *f
 ///
 /// Returns 0 in case of error or if a sub-directory (hierarchy) with the requested
 /// name already exists.
+/// ifNotExist returns a pointer to an extant sub-directory instead 0.
 /// Returns a pointer to the created sub-directory or to the top sub-directory of
 /// the hierarchy (in the above example, the returned TDirectory * always points
 /// to "a").
 
-TDirectory *TDirectoryFile::mkdir(const char *name, const char *title)
+TDirectory *TDirectoryFile::mkdir(const char *name, const char *title, Bool_t ifNotExist)
 {
    if (!name || !title || !name[0]) return nullptr;
    if (!title[0]) title = name;
    if (GetKey(name)) {
-      Error("mkdir","An object with name %s exists already",name);
-      return nullptr;
+      if(ifNotExist) return (TDirectoryFile*) GetKey(name)->ReadObj();
```

Comment:
```suggestion
      if (ifNotExist)
         return (TDirectoryFile*) GetKey(name)->ReadObj();
```
(more compliant with coding conventions).

Reply:
as an fyi, python's pathlib uses `exist_ok` for this purpose

Reply:
good suggestion

Reply:
unfortunately `exist_ok` as writte does not comply with ROOT's coding conventions :sweat_smile: let's see what @pcanal says